### DISCUSSION TITLE
acme/autocert: missing certificate error

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#1988] [Q&A]
I deployed Pocketbase in **_DigitalOcean_**. When I serve pocketbase, it gives below **ERROR**.

It started but can't access admin ui.

``http: TLS handshake error from IP:PORT: acme/autocert: missing certificate``

This is my pocketbase.service file.
```
[Unit]
Description = pocketbase

[Service]
Type           = simple
User           = root
Group          = root
LimitNOFILE    = 4096
Restart        = always
RestartSec     = 5s
StandardOutput = append:/root/pocketbase/errors.log
StandardError  = append:/root/pocketbase/errors.log
ExecStart      = /root/pocketbase/pocketbase serve --http="myDomain.com:80" --https="myDomain.com:443"

[Install]
WantedBy = multi-user.target
```

Is it cause from code or server?

**Comment by ganigeorgiev** [comment]
Hm, this looks like a failed dns check, I'm not sure.

Are you using a brand new domain - maybe its dns record is not propagate everywhere yet?

Are you able to access the Admin UI without tls (mydomain.com/_/), aka. using only the `--http` flag?

**Reply by Oggy4th** [reply]
Hello, there @ganigeorgiev 

Thanks for your reply

`--https` flag was working fine yesterday. Now it's giving below errors.

First it gives:
`2023/03/06 10:02:30 http: TLS handshake error from {IP}:{PORT}: acme/autocert: missing certificate`

Then this one:
`2023/03/06 10:02:30 http: TLS handshake error from {IP}:{PORT}: 429 urn:ietf:params:acme:error:rateLimited: Error creating new order :: too many certificates (5) already issued for this exact set of domains in the last 168 hours: mydomain.com, retry after 2023-03-06T21:43:17Z: see https://letsencrypt.org/docs/duplicate-certificate-limit/`

Also this one:
`2023/03/06 08:07:15 http: TLS handshake error from {IP}:{PORT}: acme/autocert: missing server name`

When serve with only `--http` flag, it's working.

**Reply by ganigeorgiev** [reply]
The second error (_"429 urn:ietf:params:acme:error:rateLimited: Error creating new order :: too many certificates (5) already issued for this exact set of domains in the last 168 hours:"_) explains what is the issue in your case, but I'm not sure how you ended up triggering the Let's Encrypt rate limit (we also keep a cache with the generated certificates in `pb_data/.autocert_cache` to help preventing this). Check the provided url in the error message for more details - https://letsencrypt.org/docs/duplicate-certificate-limit/.

**Reply by Oggy4th** [reply]
Below image showing files in server. 
Folder named test is old pb_data. I changed name. 

I copy pocketbase file to server after I build go. Sometimes I delete pb_data then serve pocketbase. It've been working perfectly fine until yesterday evening.
![Screenshot 2023-03-06 at 19 20 51](https://user-images.githubusercontent.com/44497354/223096505-32b4fb0d-0544-4296-bf7e-837919d421cf.png)

Also I copied old`pb_data/.autocert_cache` but It doesn't work.

I tried section **Workaround** in [https://letsencrypt.org/docs/duplicate-certificate-limit/](url).
`./pocketbase serve --http="api.mydomain.com:80" --https="api.mydomain.com:443"`

It doesn't work what I imagined.
How to implement?

Don't know what should I do, I am totally lost.

**Reply by ganigeorgiev** [reply]
@Oggy4th Please read again the error message. Somehow you've hit the Let's encrypt current rate limit of 5 certificates per week (maybe during developing you've redeployed blank pb_data multiple times?).
You can't do anything for the main domain other than wait until `2023-03-06T21:43:17Z` or use a different host pair, but this is not adjustable at the moment.

The subdomain should work as long as you've setup it properly (eg. using https://docs.digitalocean.com/products/networking/dns/how-to/add-subdomain/).

**Reply by ganigeorgiev** [reply]
>  Sometimes I delete pb_data then serve pocketbase. 

That's exactly what has triggered it. The certificates cache folder is inside the `pb_data` and everytime when you delete it and start a new project it will generate a new one. If you keep a backup of an older `pb_data/.autocert_cache` you can try to restore it.

**Reply by Oggy4th** [reply]
It works when I change `pb_data/.autocert_cache` by old `pb_data/.autocert_cache`.
